pkg/fileutils/fileutils_linux.go

```diff
@@ -0,0 +1,52 @@
+package fileutils
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/containerd/containerd/log"
+	"golang.org/x/sys/unix"
+)
+
+// GetTotalUsedFds Returns the number of used File Descriptors by
+// reading it via /proc filesystem.
+func GetTotalUsedFds() int {
+	name := fmt.Sprintf("/proc/%d/fd", os.Getpid())
```

Comment:
Surprised you used Sprintf here :)

Reply:
Haha, yeah I was doubting, but in the total scheme of things, it was slightly more readable than strconv and path.Join combined